src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnInit } from '@angular/core';
+import { StudentService } from '../../../services/student.graphql.service';
+import Student, { Gender } from '../../../../../models/student.model';
+import { ClassService } from '../../../../class/services/class.graphql.service';
+import { Class } from '../../../../../models/class.model';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-student-details-personal-info',
+  templateUrl: './student-details-personal-info.component.html',
+  styleUrls: ['./student-details-personal-info.component.scss'],
+})
+export class StudentDetailsPersonalInfoComponent implements OnInit {
+
+  student: Student;
+  classes: Class[];
+  isNewStudent: boolean;
+  idOrNew: string;
+
+  constructor(
+    private studentService: StudentService,
+    private classService: ClassService,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) { }
+
+  ngOnInit(): void {
+    if (this.route && this.route.parent) {
+      this.route.parent.params.subscribe((params) => {
+        this.idOrNew = params.idOrNew;
+      });
+    }
+
+    this.populateClasses();
+    this.populateStudent();
+  }
+
+  async populateStudent() {
+    if (this.idOrNew === '_new_') {
+      this.isNewStudent = true;
+      this.student = this.createNewStudent();
+    } else {
+      this.isNewStudent = false;
+      try {
+        this.student = await this.studentService.getById(this.idOrNew);
+      } catch (error) {
+        // TODO: implement error handling on UI
+        console.error('Error handling not implemented');
+        throw error;
+      }
+    }
+  }
+
+  createNewStudent(): Student {
+    const student: Student = new Student();
```

Comment:
This initialization can be moved into `Student` constructor

Here and below I will add comments and approve the PR, please consider to fix/comment on this review after the PR 😄